snuba/query/validation/__init__.py

```diff
@@ -0,0 +1,19 @@
+from abc import ABC, abstractmethod
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.query.expressions import Expression
+
+
+class ExpressionValidator(ABC):
+    """
+    Validates an individual expression in a Snuba logical query.
+    """
+
+    @abstractmethod
+    def validate(self, exp: Expression, schema: ColumnSet) -> None:
```

Comment:
I hope the columnset should be enough, and it would be for the validators I have inn mind. We may revisit this interface if needed.